ext/oci8/oci8_interface.c

```diff
@@ -628,28 +625,23 @@ PHP_FUNCTION(oci_lob_erase)
 	zval *tmp, *z_descriptor;
 	php_oci_descriptor *descriptor;
 	ub4 bytes_erased;
-	zend_long offset = -1, length = -1;
+	zend_long offset, length;
+	zend_bool offset_is_null = 1, length_is_null = 1;
 
-	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|ll", &z_descriptor, oci_lob_class_entry_ptr, &offset, &length) == FAILURE) {
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|l!l!", &z_descriptor, oci_lob_class_entry_ptr, &offset, &offset_is_null, &length, &length_is_null) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	if (getThis() && ZEND_NUM_ARGS() > 0 && offset < 0) {
-		php_error_docref(NULL, E_WARNING, "Offset must be greater than or equal to 0");
-		RETURN_FALSE;
-	}
-
-	if (getThis() && ZEND_NUM_ARGS() > 1 && length < 0) {
-		php_error_docref(NULL, E_WARNING, "Length must be greater than or equal to 0");
-		RETURN_FALSE;
-	}
-
-	if (!getThis() && ZEND_NUM_ARGS() > 1 && offset < 0) {
+	if (offset_is_null) {
+		offset = -1;
+	} else if (offset < 0) {
 		php_error_docref(NULL, E_WARNING, "Offset must be greater than or equal to 0");
 		RETURN_FALSE;
 	}
 
-	if (!getThis() && ZEND_NUM_ARGS() > 2 && length < 0) {
+	if (length_is_null) {
+		length = -1;
+	else if (length < 0) {
```

Comment:
This needs a closing brace for the first clause:

```
--- a/ext/oci8/oci8_interface.c
+++ b/ext/oci8/oci8_interface.c
@@ -641,7 +641,7 @@ PHP_FUNCTION(oci_lob_erase)
 
 	if (length_is_null) {
 		length = -1;
-	else if (length < 0) {
+	} else if (length < 0) {
 		php_error_docref(NULL, E_WARNING, "Length must be greater than or equal to 0");
 		RETURN_FALSE;
 	}
```

Reply:
Uhh :/